app/scripts/directives/metas/metas-actividades.js

```diff
@@ -38,247 +39,380 @@ angular.module('contractualClienteApp')
-            width: '30%',
-            headerCellClass: $scope.highlightFilteredHeader + ' text-info',
-            cellTooltip: function (row) {
-              return row.entity.valor_actividad;
+          columnDefs: [
+            {
+              field: "actividad_id",
+              displayName: "Código",
+              width: "20%",
+              headerCellClass:
+                $scope.highlightFilteredHeader + "text-center text-info",
+              cellTooltip: function (row) {
+                return row.entity.actividad_id;
+              },
             },
-            cellClass: function () {
-                  return "money";
-          }
-          }
-          ]
+            {
+              field: "actividad",
+              displayName: "Actividad",
+              width: "50%",
+              headerCellClass: $scope.highlightFilteredHeader + " text-info",
+              cellTooltip: function (row) {
+                return row.entity.actividad;
+              },
+            },
+            {
+              field: "valor_actividad",
+              displayName: "Saldo Actividad",
+              cellFilter: "currency",
+              width: "30%",
+              headerCellClass: $scope.highlightFilteredHeader + " text-info",
+              cellTooltip: function (row) {
+                return row.entity.valor_actividad;
+              },
+              cellClass: function () {
+                return "money";
+              },
+            },
+          ],
         };
 
         self.cargarMetas = function () {
-
           if ($scope.dependenciasolicitante) {
-            metasRequest.get('plan_adquisiciones/2019/'+ $scope.dependenciasolicitante.toString()).then(
-              function (res) {
-                var tempmetas = res.data.metas.actividades.filter(function(a){return a.rubro===$scope.apropiacion.RubroId}); // falta un filter por rubro
+            //TODO: Cambiar la solicitud de las metas por la URL correcta
+            const QUERY = "?query=Activo:true,RegistroPlanAdquisicionesActividadId__ActividadId__MetaId__Rubro:" +
+            $scope.apropiacion.RubroId;
+            metasRequest
+              .get(QUERY)      
+              .then(function (res) {
                 self.metas = [];
+                res.data.map((meta) => {
+                  let metaSchema = {
+                    Id: meta.RegistroPlanAdquisicionesActividadId.ActividadId
+                      .MetaId.Id,
+                    Nombre:
+                      meta.RegistroPlanAdquisicionesActividadId.ActividadId
+                        .MetaId.Nombre,
+                  };
+
+                  if (self.metas.length > 0) {
+                    self.metas.forEach((uniqueMeta) => {
+                      if (uniqueMeta.Id !== metaSchema.Id) {
+                        self.metas.push(metaSchema);
+                      }
+                    });
+                  } else {
+                    self.metas.push(metaSchema);
+                  }
```

Comment:
Aprovechando las funciones de [Array](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Array), esto se puede reescribir un poco para separar las responsabilidades y desacoplar un poco el código:

```suggestion
                self.metas = res.data.filter((meta) => {
                  // (reescribir) la condición evaluada en las lineas 96-104
                }).map( meta => {
                  // La transformación realizada en las lineas 88-94
```
(Nota: cambio sugerido, no es necesario aplicarlo)
(inclusive, el `.filter()` podría hacerse desde el query?)

Reply:
No se realiza porque tengo una confusión de cómo podría aplicar el filter en este caso, lo que se busca es que no se imprima más de una meta con el mismo id, se plantea discusión para daily